mlir/lib/Dialect/Tosa/IR/TosaOps.cpp

```diff
@@ -271,6 +271,55 @@ LogicalResult tosa::ConstOp::verify() {
   return success();
 }
 
+template <typename T>
+static LogicalResult verifyConvOpModes(T op) {
+  auto inputEType =
+      llvm::cast<ShapedType>(op.getInput().getType()).getElementType();
+
+  if (auto quantType =
+          llvm::dyn_cast<mlir::quant::UniformQuantizedType>(inputEType))
+    inputEType = quantType.getStorageType();
+
+  auto accType = op.getAccType();
+  if (inputEType.isInteger(8) && !accType.isInteger(32))
+    return op.emitOpError("accumulator type for i8 tensor is not i32");
```

Comment:
In general the dialect can allow a superset of operations to be represented when compared to the specification. I suspect these checks may need relaxing for some users of the tosa dialect. The validation pass can be used in a pipeline to more strictly check for compatibility with the specification.